internal/pkg/pkg_test.go

```diff
@@ -307,3 +310,188 @@ func TestSubpackages(t *testing.T) {
 		})
 	}
 }
+
+//nolint:scopelint
+func TestSubpackages(t *testing.T) {
+	type variants struct {
+		matcher   []SubpackageMatcher
+		recursive []bool
+		expected  []string
+	}
+
+	testCases := map[string]struct {
+		pkg   *pkgbuilder.RootPkg
+		cases []variants
+	}{
+		"remote and local nested subpackages": {
```

Comment:
Can you comment the `tree` output of the package for readability. Nested structures in test cases are tough to read.

Reply:
Added an illustration for each test case that shows the package hierarchy. I think this should make it easier to quickly see the structure.